chisel/src/cli.rs

```diff
@@ -0,0 +1,200 @@
+//! Chisel CLI
+//!
+//! This module contains the core readline loop for the Chisel CLI as well as the
+//! executable's `main` function.
+
+use chisel::{
+    prelude::{ChiselCommand, ChiselDisptacher, DispatchResult},
+    solidity_helper::SolidityHelper,
+};
+use clap::Parser;
+use foundry_cli::cmd::{forge::build::BuildArgs, LoadConfig};
+use foundry_common::evm::EvmArgs;
+use foundry_config::{
+    figment::{
+        value::{Dict, Map},
+        Metadata, Profile, Provider,
+    },
+    Config,
+};
+use rustyline::{error::ReadlineError, Editor};
+use yansi::Paint;
+
+// Loads project's figment and merges the build cli arguments into it
+foundry_config::merge_impl_figment_convert!(ChiselParser, opts, evm_opts);
+
+/// Version tag generated by vergen
+pub(crate) const VERSION_MESSAGE: &str = concat!(
+    env!("CARGO_PKG_VERSION"),
+    " (",
+    env!("VERGEN_GIT_SHA_SHORT"),
+    " ",
+    env!("VERGEN_BUILD_TIMESTAMP"),
+    ")"
+);
+
+/// Chisel is a fast, utilitarian, and verbose solidity REPL.
+#[derive(Debug, Parser)]
+#[clap(name = "chisel", version = VERSION_MESSAGE)]
+pub struct ChiselParser {
+    #[clap(flatten, next_help_heading = "BUILD OPTIONS")]
+    pub opts: BuildArgs,
+
+    #[clap(flatten, next_help_heading = "EVM OPTIONS")]
+    pub evm_opts: EvmArgs,
+
+    #[command(subcommand)]
+    pub sub: Option<ChiselParserSub>,
+}
+
+/// Chisel binary subcommands
+#[derive(clap::Subcommand, Debug)]
+pub enum ChiselParserSub {
+    /// List all cached sessions
+    List,
+    /// Load a cached session
+    Load { id: String },
+    /// View the source of a cached session
+    View { id: String },
+    /// Clear all cached chisel sessions from the cache directory
+    ClearCache,
+}
+
+#[tokio::main]
+async fn main() {
+    // Parse command args
```

Comment:
forge,cast all install a panic handler here that formats errors.

do we want this here as well and ideally convert user input panics into errors

Reply:
Done 👍